src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -255,6 +255,50 @@ void ov::npuw::LLMInferRequest::init_lora_states() {
     }
 }
 
+std::string ov::npuw::LLMInferRequest::init_pre_alloc_device() {
+    bool pre_alloc_on_npu = false;
+    const auto& prefill_compiled = m_npuw_llm_compiled_model->m_prefill_compiled;
+    for (std::size_t idx = 0; idx < prefill_compiled->m_compiled_submodels.size(); ++idx) {
+        if (prefill_compiled->submodel_device(idx) == "NPU") {
+            pre_alloc_on_npu = true;
+            break;
+        }
+    }
```

Comment:
So what's the reason for this check? I believe the idea is to.. guarantee kvcache will be allocated on NPU to make sure the 2nd model reads it with no overhead - but shouldn't we check the 2nd model devices instead?